src/Glpi/Features/ParentStatus.php

```diff
@@ -164,15 +164,48 @@ public function updateParentStatus(CommonITILObject $parentitem, array $input):
             }
         }
 
-        if (
-            !$is_set_pending
-            && !empty($this->fields['begin'])
-            && $parentitem->isStatusExists(CommonITILObject::PLANNED)
-            && (($parentitem->fields["status"] == CommonITILObject::INCOMING)
-              || ($parentitem->fields["status"] == CommonITILObject::ASSIGNED)
-              || $needupdateparent)
-        ) {
-            $input['_status'] = CommonITILObject::PLANNED;
+        if (!$is_set_pending) {
+            if (
+                !empty($this->fields['begin'])
+                && $parentitem->isStatusExists(CommonITILObject::PLANNED)
+                && (($parentitem->fields["status"] == CommonITILObject::INCOMING)
+                    || ($parentitem->fields["status"] == CommonITILObject::ASSIGNED)
+                    || $needupdateparent)
+            ) {
+                $input['_status'] = CommonITILObject::PLANNED;
+            } elseif (
+                $parentitem->fields["status"] == CommonITILObject::PLANNED
+                && in_array('begin', $this->updates)
+                && $this->isField('begin')
+            ) {
+                /** @var \DBmysql $DB */
+                global $DB;
+                $criteria = [
+                    'DISTINCT' => true,
+                    'FROM' => $this->getTable(),
+                    'WHERE' => [
+                        $parentitem->getForeignKeyField() => $parentitem->getID(),
+                        ['NOT'   => ['begin' => null]]
+                    ],
+                ];
+                $iterator = $DB->request($criteria);
+                if ($iterator->numrows() > 0) {
+                    $input['_status'] = CommonITILObject::PLANNED;
+                } elseif (
+                    $parentitem::isAllowedStatus($parentitem->fields["status"], CommonITILObject::ASSIGNED)
```

Comment:
I am not sure of this, but I think we should not rely on the current profile statuses matrix to filter available status. Indeed, I think that when a task begin date is updated, the ticket status should be changed to `PLANNED`, even if the technician is not supposed to be able to do it by itself.

@orthagh What do you think of this?

```suggestion
                    $parentitem->isStatusExists(CommonITILObject::ASSIGNED)
```

Reply:
Yes, previously, we just checked the planned status really exists for the ITILobject (Changes doesn't have this status).
The right of a technician to change the status to planned should not be checked.
This is for the UI and prevents manually changing the status field.
All shortcuts, like "adding a tech actor -> status move to in ASSIGN", "remove all attributed actor -> return to INCOMING" and the new one to move to PLANNED should be done by the framework automatically and do not impact the status field directly.